Node2Code_Engine/Compute/ApplyGroups.cs

```diff
@@ -0,0 +1,54 @@
+﻿using BH.oM.Node2Code;
+using BH.oM.Programming;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.Engine.Node2Code
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        public static List<INode> ApplyGroups(this List<INode> nodes, List<NodeGroup> groups)
```

Comment:
Could we have documentation as part of this PR? If not, can we raise an issue to add it before this becomes widely used? Just cause a few of the methods/functions in this PR may not be as obvious to users as they are to us.

For the remainder of this review, I shall simply put `Documentation?` where I think documentation is needed, but it can be remedied by an issue as per this comment and all marked as resolved if you'd prefer (I just don't want to type this out for every method 😛 )